### DISCUSSION TITLE
Does axys support direct loading of FBX file?

### DISCUSSION THREAD

**Discussion by cchuicchui** [axmolengine/axmol#869] [Q&A] [answered]
Does axys support direct loading of FBX file?

I think this is necessary since converting FBX files to C3B/C3T files cannot be done by the game alone if the game is running in a smart phone.

**Comment by aismann** [comment]
Plz look here:
[axys](https://github.com/axmol1/axys/tree/18076d09f039b5ebebcb411332f8e95f1c3a43fd)/[tools](https://github.com/axmol1/axys/tree/18076d09f039b5ebebcb411332f8e95f1c3a43fd/tools)/[fbx-conv](https://github.com/axmol1/axys/tree/18076d09f039b5ebebcb411332f8e95f1c3a43fd/tools/fbx-conv)/README.md

fbxconverter version 0.7 It is for cocos2d-x after version v3.3
Convert .fbx to .c3t used by cocos2d-x. Thanks to libgdx fbx-conv. We use their code and do some change.
You can get the source code from the address below: https://github.com/cocos2d-x/fbx-conv


Direct loading is not supported.

**Reply by cchuicchui** [reply]
Any chance of adding direct loading of fbx to axys? Is it very complicated in terms of coding?

A simple implementation would be converting fbx-conv from a command line utility to API such that we can call this API to dynamically convert fbx to c3b directly inside the iphone. Then we read the c3b file.

**Reply by aismann** [reply]
> Any chance of adding direct loading of fbx to axys? Is it very complicated in terms of coding?
> 
> A simple implementation would be converting fbx-conv from a command line utility to API such that we can call this API to 
> dynamically convert fbx to c3b directly inside the iphone. Then we read the c3b file.

You or everybody can write an extension or a PR.  (please mark it as answer)

**Comment by cchuicchui** [comment, marked as answer]
Do u mean i write it myself?

**Reply by aismann** [reply]
> Do u mean i write it myself?

Yes. Why not?

**Reply by cchuicchui** [reply]
I will try. Need sometime to get familiar with how to do PR.

**Reply by aismann** [reply]
Plz Check the license also.
axys is MIT.

**Reply by cchuicchui** [reply]
FBX SDK has been used in cocos2dx long time ago. How come adding an API on top of the command which serves the same function would violate the license?

**Reply by aismann** [reply]
It was only a hint.

**Reply by aismann** [reply]
Maybe this is also an interesting sides.
https://www.autodesk.com/developer-network/platform-technologies/fbx-sdk-2020-2-1
https://github.com/cocos/FBX-glTF-conv

**Reply by halx99** [reply]
PR guide: https://github.com/axmolengine/axmol/discussions/411
